benches/frequency-map.rs

```diff
@@ -0,0 +1,1130 @@
+        ( 0, 1),
+        ( 1, 1),
+        ( 2, 1),
+        ( 3, 1),
+        ( 4, 1),
+        ( 5, 1),
+        ( 6, 1),
+        ( 7, 1),
+        ( 8, 1),
+        ( 9, 1),
+        (10, 1),
+        (11, 1),
+        (12, 1),
+        (13, 1),
+        (14, 1),
+        (15, 1),
+        (16, 1),
+        (17, 1),
+        (18, 1),
+        (19, 1),
+        (20, 1),
+        (21, 1),
+        (22, 1),
+        (23, 1),
+        (24, 1),
+        (25, 1),
+        (26, 1),
+        (27, 1),
+        (28, 1),
+        (29, 1),
+        (30, 1),
+        (31, 1),
+        (32, 1),
+        (33, 1),
+        (34, 1),
+        (35, 1),
+        (36, 1),
+        (37, 1),
+        (38, 1),
+        (39, 1),
+        (40, 1),
+        (41, 1),
+        (42, 1),
+        (43, 1),
+        (44, 1),
+        (45, 1),
+        (46, 1),
+        (47, 1),
+        (48, 1),
+        (49, 1),
+        (50, 1),
+        (51, 1),
+        (52, 1),
+        (53, 1),
+        (54, 1),
+        (55, 1),
+        (56, 1),
+        (57, 1),
+        (58, 1),
+        (59, 1),
+        (60, 1),
+        (61, 1),
+        (62, 1),
+        (63, 1),
+        (64, 1),
+        (65, 1),
+        (66, 1),
+        (67, 1),
+        (68, 1),
+        (69, 1),
+        (70, 1),
+        (71, 1),
+        (72, 1),
+        (73, 1),
+        (74, 1),
+        (75, 1),
+        (76, 1),
+        (77, 1),
+        (78, 1),
+        (79, 1),
+        (80, 1),
+        (81, 1),
+        (82, 1),
+        (83, 1),
+        (84, 1),
+        (85, 1),
+        (86, 1),
+        (87, 1),
+        (88, 1),
+        (89, 1),
+        (90, 1),
+        (91, 1),
+        (92, 1),
+        (93, 1),
+        (94, 1),
+        (95, 1),
+        (96, 1),
+        (97, 1),
+        (98, 1),
+        (99, 1),
+    ];
+
+    pub(crate) const I32_RANGE_EMPTY : std_ops::Range<i32> = 0..0;
+
+    pub(crate) const I32_RANGE_SMALL : std_ops::Range<i32> = 0..10;
+
+    pub(crate) const I32_RANGE_MEDIUM : std_ops::Range<i32> = 0..100;
+
+    pub(crate) const I32_RANGE_MEDIUM_2 : std_ops::Range<i32> = 20..60;
+    pub(crate) const I32_RANGE_MEDIUM_3 : std_ops::Range<i32> = 40..80;
+
+    pub(crate) const I32_RANGE_LARGE : std_ops::Range<i32> = 0..10000;
+
+    pub(crate) const I32_RANGE_LARGE_2 : std_ops::Range<i32> = 2000..6000;
+    pub(crate) const I32_RANGE_LARGE_3 : std_ops::Range<i32> = 4000..8000;
+}
+
+fn BENCHMARK_new(c : &mut Criterion) {
+    let id = format!("`frequency_map::new()`");
+
+    c.bench_function(&id, |b| {
+        b.iter(|| {
+            let fm : FrequencyMap<String> = FrequencyMap::new();
+
+            let _ = black_box(fm);
+        })
+    });
+}
+
+fn BENCHMARK_with_capacity_SMALL(c : &mut Criterion) {
+    const INITIAL_CAPACITY : usize = 10;
+    let id = format!("`frequency_map::with_capacity({INITIAL_CAPACITY})`");
+
+    c.bench_function(&id, |b| {
+        b.iter(|| {
+            let fm : FrequencyMap<String> = FrequencyMap::with_capacity(INITIAL_CAPACITY);
```

Comment:
```suggestion
            let fm : FrequencyMap<String> = FrequencyMap::with_capacity(black_box(INITIAL_CAPACITY));
```

The inputs should be passed through a `black_box` to inhibit the optimizer from inlining this.